octoprint_mrbeam/gcodegenerator/img2gcode.py

```diff
@@ -477,8 +491,16 @@ def generate_gcode(self, imgArray, xMM, yMM, wMM, hMM, file_id):
                     img_data["id"], img_data["x"], img_data["y"], size[0], size[1]
                 )
             )
+
+            # TODO improvement: find first non-white pixel from lower left
+            x_start = x_off
+            y_start = (
+                y_off - height_px * self.beam
+            )  # lower left of the image, but with safety whitespace around the content
             gc = self._get_gcode_g0(
-                x=x_off, y=y_off, comment="; Move to start ({},{})".format(x_off, y_off)
+                x=x_start,
+                y=y_start,
+                comment="; Move to start ({},{})".format(x_start, y_start),
             )
```

Comment:
We do need this to ramp up the laser driver. 
Basically this is a bugfix. The change is only in the coordinates - and yes, it still could be improved by figuring out the real first pixel of the image. But I skipped further optimization as our resources are sparse and there is no business benefit. Additionally code stays more modular.

Reply:
Yeah but I mean, when we do the image scan it's already creating a move command before lasering the first pixels. To me this could even move the laserhead to a white line?

Reply:
It might not even be starting on the correct line in the case of clusters.